### PR TITLE
Pairs::peek

### PR DESCRIPTION
**Pull request by CAD97**
How does `next_back` work? Would a `peek_back` be useful to have (for completeness?)?

### CODE REVIEW

File: pest/src/iterators/pairs.rs [hunk 1]
```diff
@@ -167,6 +167,15 @@ impl<'i, R: RuleType> Pairs<'i, R> {
         tokens::new(self.queue, self.input, self.start, self.end)
     }
 
+    #[inline]
+    pub fn peek(&mut self) -> Option<<Self as Iterator>::Item> {
```
Comment (dragostis) [inline pest/src/iterators/pairs.rs]: Why use `<Self as Iterator>::Item` instead of `Pair`?

```suggestion
    pub fn peek(&mut self) -> Option<Pair> {
```

Comment (dragostis) [inline pest/src/iterators/pairs.rs]: Should probably have docs with an example.

**Comment by CAD97** [comment]
bors: r=dragostis

**Comment by CAD97** [comment]
Whoops on the signature 😅 
Blame mobile that I didn't realize that CI had _already_ reported that...

bors: r=dragostis

**Comment by CAD97** [comment]
bors: cancel

Actually, going to add minimal docs real quickly since I've got it pulled up

**Comment by CAD97** [comment]
... And fix a huge oversight that I had made `peek` take `&mut self` instead of `&self`

**Comment by CAD97** [comment]
Alright, this time we should be good. Good enough to merge, we can iterate on the docs from here.

bors: r=dragostis

**Pull request merged**